aws/rust-runtime/aws-runtime/src/auth/sigv4.rs

```diff
@@ -152,47 +158,33 @@ impl Sign for SigV4Signer {
         runtime_components: &RuntimeComponents,
         config_bag: &ConfigBag,
     ) -> Result<(), BoxError> {
-        let operation_config =
-            Self::extract_operation_config(auth_scheme_endpoint_config, config_bag)?;
-        let request_time = runtime_components.time_source().unwrap_or_default().now();
-
         if identity.data::<Credentials>().is_none() {
-            return Err(SigV4SigningError::WrongIdentityType(identity.clone()).into());
+            return Err(SigV4SigningError {
+                kind: ErrorKind::WrongIdentityType(identity.clone()),
+            }
+            .into());
         };
 
+        let operation_config =
+            Self::extract_operation_config(auth_scheme_endpoint_config, config_bag)?;
+        let time_source = runtime_components.time_source().unwrap_or_default();
+
         let settings = Self::settings(&operation_config);
         let signing_params =
-            Self::signing_params(settings, identity, &operation_config, request_time)?;
+            Self::signing_params(settings, identity, &operation_config, time_source.now())?;
 
-        let (signing_instructions, _signature) = {
-            // A body that is already in memory can be signed directly. A body that is not in memory
-            // (any sort of streaming body or presigned request) will be signed via UNSIGNED-PAYLOAD.
-            let signable_body = operation_config
+        let signing_params = SigningParams::V4(signing_params);
+        let mut signing_package = SigningPackage::builder().signing_params(&signing_params);
+        signing_package.set_payload_override(
+            operation_config
                 .signing_options
                 .payload_override
                 .as_ref()
-                // the payload_override is a cheap clone because it contains either a
-                // reference or a short checksum (we're not cloning the entire body)
-                .cloned()
-                .unwrap_or_else(|| {
-                    request
-                        .body()
-                        .bytes()
-                        .map(SignableBody::Bytes)
-                        .unwrap_or(SignableBody::UnsignedPayload)
-                });
-
-            let signable_request = SignableRequest::new(
-                request.method(),
-                request.uri(),
-                request.headers().iter(),
-                signable_body,
-            )?;
-            sign(signable_request, &SigningParams::V4(signing_params))?
-        }
-        .into_parts();
+                .cloned(),
+        );
+
+        let (_, _signature) = request.sign_with(&signing_package.build()?)?.into_parts();
 
-        // If this is an event stream operation, set up the event stream signer
```

Comment:
Why remove the comment?

Reply:
Oh removal was by mistake. Will restore it.